mesonbuild/msetup.py

```diff
@@ -256,6 +266,8 @@ def _generate(self, env: environment.Environment, capture: bool, vslite_ctx: T.O
                 # read from a pipe and wrote into a private file.
                 self.options.cross_file = env.coredata.cross_files
                 self.options.native_file = env.coredata.config_files
+                self.options.cross_program = [f'{k[0]}={v[1]}' for k, v in env.coredata.get_program_overrides(MachineChoice.HOST).items()]
+                self.options.native_program = [f'{k[0]}={v[1]}' for k, v in env.coredata.get_program_overrides(MachineChoice.BUILD).items()]
```

Comment:
In terms of readability this feels somehow inscrutable.

It also unfortunately doesn't actually get dumped to cmd_line.txt at all.

Reply:
There was a bug. These shouldn't have been indexed. Let me know what you think